app/src/main/java/de/danoeh/antennapod/preferences/PreferenceController.java

```diff
@@ -433,9 +436,17 @@ public void onResume() {
         setParallelDownloadsText(UserPreferences.getParallelDownloads());
         setEpisodeCacheSizeText(UserPreferences.getEpisodeCacheSize());
         setDataFolderText();
+        GpodnetPreferences.setSyncAttemptListener(() -> ui.getActivity().runOnUiThread(
```

Comment:
It does essentially the same, except that it's more direct (and doesn't get triggered on other unrelated preference changes). Do you think it gets in the way of code readability?